File: pset6/caesar.py
import sys
import cs50

def main() :
    if len(sys.argv) != 2 :
        print("USAGE: python caesar.py k")
        return 1

    k = int(sys.argv[1])
    k = k % 26

    # store plaintext
    print("Plaintext: ", end="")
    ptxt = cs50.get_string()
    print("Ciphertext: ", end="")

    ciphertext  = ""
    for i in range(len(ptxt)) :
        if ptxt[i].isupper() :
            if (ord(ptxt[i]) + k - 90) > 0 :
                print(chr(64 + ord(ptxt[i]) + k - 90), end="")
            else :
                print(chr(ord(ptxt[i]) + k), end="")
        elif ptxt[i].islower() :
            if (ord(ptxt[i]) + k - 122) > 0 :
                print(chr(96 + ord(ptxt[i]) + k - 122), end="")
            else :
                print(chr(ord(ptxt[i]) + k), end="")
        else :
            print(ptxt[i], end="")

    print("")

if __name__ == "__main__" :
    main()

File: pset2/caeser/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("USAGE: ./caeser k\n");
        return 1;
    }
    int n = 0, i;
    // storing key as integer
    for (i = 0; argv[1][i]; i++)
    {
        n = n * 10 + (int)argv[1][i] - 48;
    }

    //making key smaller
    n %= 26;

    //taking original message from user
    printf("plaintext: ");
    string ptxt = get_string();

    for (i = 0; ptxt[i]; i++)
    {
        //checking if character is alphabet
        if (isalpha(ptxt[i]))
        {
            //encrypting message using key
            if (islower(ptxt[i]))
            {
                ptxt[i] = ((ptxt[i] + n - 122) > 0 ? 96 + (ptxt[i] + n - 122) : ptxt[i] + n);
            }
            else
            {
                ptxt[i] = ((ptxt[i] + n - 90) > 0 ? 64 + (ptxt[i] + n - 90) : ptxt[i] + n);
            }
        }
    }
    printf("ciphertext: %s\n", ptxt);
}